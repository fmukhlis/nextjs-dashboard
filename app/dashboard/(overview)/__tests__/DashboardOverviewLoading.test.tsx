import { render } from "@testing-library/react";
import OverviewLoading from "../loading";

describe("DashboardOverviewLoading", () => {
  // Snapshot
  it("renders DashboardOverviewLoading unchanged", () => {
    const { container } = render(<OverviewLoading />);
    expect(container).toMatchSnapshot();
  });
});
