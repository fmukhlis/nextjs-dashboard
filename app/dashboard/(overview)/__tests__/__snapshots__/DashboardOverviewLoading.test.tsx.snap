// Jest Snapshot v1, https://jestjs.io/docs/snapshot-testing

exports[`DashboardOverviewLoading renders DashboardOverviewLoading unchanged 1`] = `
<div>
  <div
    class="before:absolute before:inset-0 before:-translate-x-full before:animate-[shimmer_2s_infinite] before:bg-gradient-to-r before:from-transparent before:via-white/60 before:to-transparent relative mb-4 h-8 w-36 overflow-hidden rounded-md bg-gray-100"
  />
  <div
    class="grid gap-6 sm:grid-cols-2 lg:grid-cols-4"
  >
    <div
      class="before:absolute before:inset-0 before:-translate-x-full before:animate-[shimmer_2s_infinite] before:bg-gradient-to-r before:from-transparent before:via-white/60 before:to-transparent relative overflow-hidden rounded-xl bg-gray-100 p-2 shadow-sm"
    >
      <div
        class="flex p-4"
      >
        <div
          class="h-5 w-5 rounded-md bg-gray-200"
        />
        <div
          class="ml-2 h-6 w-16 rounded-md bg-gray-200 text-sm font-medium"
        />
      </div>
      <div
        class="flex items-center justify-center truncate rounded-xl bg-white px-4 py-8"
      >
        <div
          class="h-7 w-20 rounded-md bg-gray-200"
        />
      </div>
    </div>
    <div
      class="before:absolute before:inset-0 before:-translate-x-full before:animate-[shimmer_2s_infinite] before:bg-gradient-to-r before:from-transparent before:via-white/60 before:to-transparent relative overflow-hidden rounded-xl bg-gray-100 p-2 shadow-sm"
    >
      <div
        class="flex p-4"
      >
        <div
          class="h-5 w-5 rounded-md bg-gray-200"
        />
        <div
          class="ml-2 h-6 w-16 rounded-md bg-gray-200 text-sm font-medium"
        />
      </div>
      <div
        class="flex items-center justify-center truncate rounded-xl bg-white px-4 py-8"
      >
        <div
          class="h-7 w-20 rounded-md bg-gray-200"
        />
      </div>
    </div>
    <div
      class="before:absolute before:inset-0 before:-translate-x-full before:animate-[shimmer_2s_infinite] before:bg-gradient-to-r before:from-transparent before:via-white/60 before:to-transparent relative overflow-hidden rounded-xl bg-gray-100 p-2 shadow-sm"
    >
      <div
        class="flex p-4"
      >
        <div
          class="h-5 w-5 rounded-md bg-gray-200"
        />
        <div
          class="ml-2 h-6 w-16 rounded-md bg-gray-200 text-sm font-medium"
        />
      </div>
      <div
        class="flex items-center justify-center truncate rounded-xl bg-white px-4 py-8"
      >
        <div
          class="h-7 w-20 rounded-md bg-gray-200"
        />
      </div>
    </div>
    <div
      class="before:absolute before:inset-0 before:-translate-x-full before:animate-[shimmer_2s_infinite] before:bg-gradient-to-r before:from-transparent before:via-white/60 before:to-transparent relative overflow-hidden rounded-xl bg-gray-100 p-2 shadow-sm"
    >
      <div
        class="flex p-4"
      >
        <div
          class="h-5 w-5 rounded-md bg-gray-200"
        />
        <div
          class="ml-2 h-6 w-16 rounded-md bg-gray-200 text-sm font-medium"
        />
      </div>
      <div
        class="flex items-center justify-center truncate rounded-xl bg-white px-4 py-8"
      >
        <div
          class="h-7 w-20 rounded-md bg-gray-200"
        />
      </div>
    </div>
  </div>
  <div
    class="mt-6 grid grid-cols-1 gap-6 md:grid-cols-4 lg:grid-cols-8"
  >
    <div
      class="before:absolute before:inset-0 before:-translate-x-full before:animate-[shimmer_2s_infinite] before:bg-gradient-to-r before:from-transparent before:via-white/60 before:to-transparent relative w-full overflow-hidden md:col-span-4"
    >
      <div
        class="mb-4 h-8 w-36 rounded-md bg-gray-100"
      />
      <div
        class="rounded-xl bg-gray-100 p-4"
      >
        <div
          class="sm:grid-cols-13 mt-0 grid h-[410px] grid-cols-12 items-end gap-2 rounded-md bg-white p-4 md:gap-4"
        />
        <div
          class="flex items-center pb-2 pt-6"
        >
          <div
            class="h-5 w-5 rounded-full bg-gray-200"
          />
          <div
            class="ml-2 h-4 w-20 rounded-md bg-gray-200"
          />
        </div>
      </div>
    </div>
    <div
      class="before:absolute before:inset-0 before:-translate-x-full before:animate-[shimmer_2s_infinite] before:bg-gradient-to-r before:from-transparent before:via-white/60 before:to-transparent relative flex w-full flex-col overflow-hidden md:col-span-4"
    >
      <div
        class="mb-4 h-8 w-36 rounded-md bg-gray-100"
      />
      <div
        class="flex grow flex-col justify-between rounded-xl bg-gray-100 p-4"
      >
        <div
          class="bg-white px-6"
        >
          <div
            class="flex flex-row items-center justify-between border-b border-gray-100 py-4"
          >
            <div
              class="flex items-center"
            >
              <div
                class="mr-2 h-8 w-8 rounded-full bg-gray-200"
              />
              <div
                class="min-w-0"
              >
                <div
                  class="h-5 w-40 rounded-md bg-gray-200"
                />
                <div
                  class="mt-2 h-4 w-12 rounded-md bg-gray-200"
                />
              </div>
            </div>
            <div
              class="mt-2 h-4 w-12 rounded-md bg-gray-200"
            />
          </div>
          <div
            class="flex flex-row items-center justify-between border-b border-gray-100 py-4"
          >
            <div
              class="flex items-center"
            >
              <div
                class="mr-2 h-8 w-8 rounded-full bg-gray-200"
              />
              <div
                class="min-w-0"
              >
                <div
                  class="h-5 w-40 rounded-md bg-gray-200"
                />
                <div
                  class="mt-2 h-4 w-12 rounded-md bg-gray-200"
                />
              </div>
            </div>
            <div
              class="mt-2 h-4 w-12 rounded-md bg-gray-200"
            />
          </div>
          <div
            class="flex flex-row items-center justify-between border-b border-gray-100 py-4"
          >
            <div
              class="flex items-center"
            >
              <div
                class="mr-2 h-8 w-8 rounded-full bg-gray-200"
              />
              <div
                class="min-w-0"
              >
                <div
                  class="h-5 w-40 rounded-md bg-gray-200"
                />
                <div
                  class="mt-2 h-4 w-12 rounded-md bg-gray-200"
                />
              </div>
            </div>
            <div
              class="mt-2 h-4 w-12 rounded-md bg-gray-200"
            />
          </div>
          <div
            class="flex flex-row items-center justify-between border-b border-gray-100 py-4"
          >
            <div
              class="flex items-center"
            >
              <div
                class="mr-2 h-8 w-8 rounded-full bg-gray-200"
              />
              <div
                class="min-w-0"
              >
                <div
                  class="h-5 w-40 rounded-md bg-gray-200"
                />
                <div
                  class="mt-2 h-4 w-12 rounded-md bg-gray-200"
                />
              </div>
            </div>
            <div
              class="mt-2 h-4 w-12 rounded-md bg-gray-200"
            />
          </div>
          <div
            class="flex flex-row items-center justify-between border-b border-gray-100 py-4"
          >
            <div
              class="flex items-center"
            >
              <div
                class="mr-2 h-8 w-8 rounded-full bg-gray-200"
              />
              <div
                class="min-w-0"
              >
                <div
                  class="h-5 w-40 rounded-md bg-gray-200"
                />
                <div
                  class="mt-2 h-4 w-12 rounded-md bg-gray-200"
                />
              </div>
            </div>
            <div
              class="mt-2 h-4 w-12 rounded-md bg-gray-200"
            />
          </div>
        </div>
        <div
          class="flex items-center pb-2 pt-6"
        >
          <div
            class="h-5 w-5 rounded-full bg-gray-200"
          />
          <div
            class="ml-2 h-4 w-20 rounded-md bg-gray-200"
          />
        </div>
      </div>
    </div>
  </div>
</div>
`;
